docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
+It has multiple methods for separate features.
+
+**Implementation**
+
+The sequence by which the ManPage class handles the `man` command is as follows:
+1) The `Parser` class extracts the other fields of the user input, and calls upon the `checkUserInput()` method
+   which calls upon the `checkManPageInput()`.
+
+2) The method then uses the methods in the ManPage class to print out the requested feature.
+
+`ManPage#printManPage()` - Prints the Command Summary.
+
+`ManPage#printManPageAdd()` - Prints description of how to use the add feature.
+
+`ManPage#printManPageList()` - Prints description of how to use the list feature.
+
+`ManPage#printManPageEdit()` - Prints description of how to use the edit feature.
+
+`ManPage#printManPageFind()` - Prints description of how to use the find feature.
+
+`ManPage#printManPageDelete()` - Prints description of how to use the delete feature.
+
+`ManPage#printManPageCalculator()` - Prints description of how to use the grade calculator.
+
+`ManPage#printManPageUpdateGrade()` - Prints description of how to input grades.
+
+`ManPage#printManPageExit()` - Prints description of how to use the exit feature.
+
+![ManPage](diagrams/ManPage.png)
+
+**Design Considerations**
+
+**Alternative 1** : Prints the manual page individually as features. 
+Pros: Avoids wordy and long manual page when command is man, ability to have more description
+for specific features. Avoids long methods. 
+Cons: Extra step of typing another command to understand a feature well.
+
+**Alternative 2**: Combining all feature descriptions in a single manual page. 
+Pros: Avoid extra step of typing another command to understand a feature well.
+Cons: Wordy and long manual page when command is man, user has to scroll through
+a long manual page to find what they need. Long method.
+
+### 3.2. Parser
+
+The `Parser` is a class that handles the commands keyed in by the user.  
+It has multiple checks to ensure that the user keys in valid commands. 
+
+<!---elaborate more on the class, still unsure of the format...--->
+<!---insert other commands here, still unsure of the format...(should it all be under Parser class or separate?)--->
+
+### 3.3. Add Modules
+
+`add` command:
+
+The `add` command is used by the user to add a module or multiple modules at once.
+
+The sequence by which the Parser class handles the `add` command is as follows:
+1) The `Parser` class extracts the necessary fields from the user input, and calls upon the `addModule`
+   method in the `ModuleList` class in a loop.
+2) The method then checks the type of module that is being added, then calls the relevant
+   constructor (`Core`, `UE`, `GE`, `Internship`).
+3) The `addModule` method returns the module that is added to the `Parser` class.
+4) The `getModuleListSize` method is called from `ModuleList` class.
+5) The `printAddedModule` method is called from the `Print` class to display the result to the user.
```

Comment:
It may be better if the point form is used in a way such that it is properly seperated
```suggestion
1.  The `Parser` class extracts the necessary fields from the user input, and calls upon the `addModule` method in the `ModuleList` class in a loop.
2. The method then checks the type of module that is being added, then calls the relevant constructor (`Core`, `UE`, `GE`, `Internship`).
3. The `addModule` method returns the module that is added to the `Parser` class.
4. The `getModuleListSize` method is called from `ModuleList` class.
5. The `printAddedModule` method is called from the `Print` class to display the result to the user.
```